### ISSUE TITLE
Pod recreatd after scale up persistence if serverconfig updated

### ISSUE THREAD

**Issue by liquanstan** [rabbitmq/cluster-operator#1428]
## Describe the bug
Pod will be killed and recreated after scale up the spec.persistence.storage. 
Only before i change spec.rabbitmq.additionalConfig or advancedConfig or limit.memory , which result in the serverconfig changed.

## To Reproduce
1. Deploy the rabbitmq-cluster-operator;
2. Apply a rabbitmqcluster yaml in k8s cluster;
3. Edit the rabbitmqcluster, change limit.memory;
4. Wait all pods recreating and running;
5. Edit the rabbitmqcluster, scale up spec.persistence.storage;
6. Pods recreating.

**Expected behavior**
PVC scaled up without Pods recreating.

## Version and environment information
- RabbitMQ: 3.9.17
- RabbitMQ Cluster Operator: 2.4.0
- Kubernetes: 1.19.12
 
## Additional context
By analyzing the controller, i think problem lies here:
https://github.com/rabbitmq/cluster-operator/blob/main/internal/scaling/scaling.go#L67-L71 and https://github.com/rabbitmq/cluster-operator/blob/main/controllers/reconcile_rabbitmq_configurations.go#L88-L104 

Scaling up persistence will delete statefulset, which will clean the stsRestartAnnotation Annotation in spec.template;
If stsRestartAnnotation is empty, any reconcile will lead to stsRestartAnnotation added to spec.template;
Only empty serverConfAnnotation can prevent pervious context from happening.

Can we just delete the "delete statefulset when scaling up persistence" code?
VolumeClaimTemplates may be inconsistent with pvc in environment after scaling up, but that seems less important.
I don't think recreating pod when  just scaling up persistent is necessary.

**Comment by Zerpet** [comment]
> Can we just delete the "delete statefulset when scaling up persistence" code?

No, that code should be there. Having an inconsistent `VolumeClaimTemplates` with what the actual PVCs is important.

We could preserve the `lastRestartAt` annotation in the StatefulSet and bring it over to the "new" StatefulSet.

**Comment by liquanstan** [comment]
@Zerpet Thank you for your response.
Want to learn a little more, as far as i know, Keeping a consistent `VolumeClaimTemplates` means Creating correct PVC when scaling up replicas and get correct volume capacity in https://github.com/rabbitmq/cluster-operator/blob/main/internal/scaling/scaling.go#L118-L130.
Anything else?

Since we maintain serval operators with statefulset and pvc, it seems like a common logic to deleting statefulset when scaling up pvcs.
